### ISSUE TITLE
Cant get Monitor Assistant to Direct Connect to installed agent.  Always connected assistant works fine.

### ISSUE THREAD

**Issue by jobrien2001** [Ylianst/MeshCentral#4861]
**Describe your issue**
Hello,

I have a problem direct connect the monitor assistant to the background agent.  The always connected assistant works fine

1- I install the background agent 
2- Download the monitor only version of the asistant
3- Run and always disconnected, remote login to desktop with agent and monitor doesnt show connected nor avatar, can request help.
4- Switch from direct connect to server and it works.
5- Download msh file with ?debug=1, copy to same folder where assistant resides.
6- Run and always disconnected, remote login to desktop with agent and monitor doesnt show connected nor avatar, can request help. 
7- Switch from direct connect to server and it works. 
8 - copied assistant to program folder where agent resides. same outcome.

Note: switching from direct connect to server creates another device on my server.

What am i doing wrong?

Thanks

**Screenshots**
![image](https://user-images.githubusercontent.com/12868319/208230890-6c1ae8d9-488c-4c53-be8b-c2aae5b30e8c.png)

**Server Software (please complete the following information):**
 - OS: Debian 11
 - Virtualization: Docker
 - Network: reverse proxy
 - Version: 1.1.1
 - Node: 18.12.1

**Remote Device (please complete the following information):**
 - Device: Laptop
 - Windows 11 22H2
 - Network: Remote over WAN
 - Current Core Version: ??

**Your config.json file**

```
{
    "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
    "settings": {
        "cert": "remote.acmecomputer.com",
        "_WANonly": true,
        "_LANonly": true,
        "_sessionKey": "xxxxxxxx",
        "port": 443,
        "aliasPort": 443,
        "redirPort": 80,
        "redirAliasPort": 80,
        "agentLogDump": true,
        "AgentPong": 300,
        "TLSOffload": "NginxProxyManager",
        "SelfUpdate": false,
        "AllowFraming": "false",
        "WebRTC": "true",
        "_webPush": {
            "_email": ""
        },
        "desktopMultiplex": true,
        "maxInvalidLogin": {
            "time": 10,
            "count": 5,
            "coolofftime": 3
        },
        "maxInvalid2fa": {
            "time": 10,
            "count": 3,
            "coolofftime": 3
        },
        "plugins": {
            "enabled": true
        }
    },
    "domains": {
        "": {
            "title": "acme Computer Remote",
            "_title1": "acme Computer Remote",
            "_title2": "acme Computer Remote",
            "titlePicture": "titlePicture.png",
            "loginPicture": "loginPicture2.png",
            "_welcomeText": "",
            "_welcomePicture": "",
            "_welcomePictureFullScreen": "",
            "meshMessengerTitle": "acme Computer Messenger",
            "_minify": true,
            "NewAccounts": "false",
            "_userNameIsEmail": true,
            "hide": 4,
            "loginfooter": "",
            "certUrl": "https://NginxProxyManager:4443",
            "passwordRequirements": {
                "min": 6,
                "upper": 1,
                "numeric": 1,
                "force2factor": true,
                "email2factor": false,
                "sms2factor": true,
                "otp2factor": false,
                "banCommonPasswords": true,
                "loginTokens": false,
                "backupcode2factor": false,
                "single2factorWarning": false,
                "allowaccountreset": false
            },
            "twoFactorCookieDurationDays": 5,
            "agentInviteCodes": true,
            "geoLocation": false,
            "consentMessages": {
                "Title": "Remote Support Request",
                "Desktop": "{0} is requesting to remote desktop control this computer. Click Allow to grant access",
                "consentTimeout": 60,
                "_autoAcceptOnTimeout": true
            },
            "notificationMessages": {
                "title": "acme Computer",
                "desktop": "{0} started a remote desktop session."
            },
            "agentCustomization": {
                "displayName": "acme Computer® Remote Service",
                "description": "acme Computer® Remote Service",
                "companyName": "acme Computer®",
                "serviceName": "acme Computer Remote",
                "image": "agentlogo.png",
                "installText": "Click the install button bellow to install the remote management software.  When installed, the software runs in the background allowing this computer to be managed and controlled by a remote administrator from acme Computer",
                "fileName": "acmecomputerremoteagent"
            },
            "agentFileInfo": {
                "fileDescription": "acme Computer® Remote Agent",
                "internalName": "acme Computer® Remote Agent",
                "legalCopyright": "acme Computer®",
                "originalFilename": "acmecomputerremoteagent",
                "productName": "acme Computer® Remote Agent",
                "icon": "agent.ico"
            },
            "assistantCustomization": {
                "title": "acme Computer® Assistant",
                "fileName": "acmecomputerremoteassistant",
                "image": "asistantlogox256.png"
            },
            "androidCustomization": {
                "title": "acme Computer Remote Agent",
                "subtitle": "Remote Access",
                "image": "asistantlogo.png"
            },
           "userConsentFlags": {
                "desktopprivacybar": "false"
            },
            "limits": {
                "MaxSingleUserSessions": 5
            },
            "smtp": {
                "title": "acmecomputer",
                "description": "Connects MeshCentral to a SMTP email server, allows MeshCentral to send email messages for 2FA or user notification.",
                "host": "mail.acmecomputer.com",
                "port": 465,
                "from": "notifications@acmecomputer.com",
                "user": "notifications@acmecomputer.com",
                "pass": "xxxxxxx",
                "tls": true,
                "tlscertcheck": true,
                "_tlsstrict": true,
                "_verifyemail": true,
                "emailDelaySeconds": 60
            }

        }
    },
    "_letsencrypt": {
        "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
        "_email": "myemail@mydomain.com",
        "_names": "myserver.mydomain.com",
        "production": false
    },
    "sms": {
        "title": "SMS provider",
        "description": "Connects MeshCentral to a SMS text messaging provider, allows MeshCentral to send SMS messages for 2FA or user notification.",
        "provider": "twilio",
        "sid": "xxxxxxxx",
        "auth": "xxxxxxxxxxx",
        "from": "xxxxxxxx"
    },
    "messaging": {
        "telegram": {
            "apiid": xxxxxxxx,
            "apihash": "xxxxxxxx",
            "bottoken": "xxxxxxxx:xxxxxxxx",
            "_proxy": {
                "ip": "NginxProxyManager",
                "port": 4443
            }
        }
    }
}
```
Assistant log, note i switch manually connect and it works. switch back to direct and bust

```
02:18:AM.0696: ***** Starting MeshCentral Assistant *****
02:18:AM.0762: Version 0.1.8175.21599
02:18:AM.0762: Set TLS 1.2
02:18:AM.0762: InitializeComponent()
02:18:AM.1236: Checking for embedded MSH file
02:18:AM.1393: Check for built-in agent
02:18:AM.1393: Starting built-in agent
02:18:AM.1710: MCAgent: LoadAgentCertificate() - Loaded existing certificate from registry.
02:18:AM.1710: MCAgent: MSH MeshID: xxxxxxxxxxxxxxxxxxxxxxxxx
02:18:AM.1710: MCAgent: MSH ServerID: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
02:18:AM.1710: MCAgent: MSH MeshServer: wss://remote.acmecomputer.com/agent.ashx
02:18:AM.1710: Get list of background agents
02:18:AM.1866: Found 1 background agent(s)
02:18:AM.2244: MainForm_Load()
02:18:AM.2712: connectToAgent ~
02:21:AM.0411: agentSelection_Click acmeComputerRemote
02:21:AM.0411: connectToAgent acmeComputerRemote
02:21:AM.0504: Agent: ConnectPipe()
02:21:AM.0504: Agent_onStateChanged 1, 0, acmeComputerRemote
02:21:AM.0504: Agent: SendCommand cmd="register", value="xxxx\xxxx"
02:21:AM.0627: Agent: SendCommand cmd="sessions", value=""
02:21:AM.0627: Agent: ConnectPipe() - BeginRead
02:21:AM.0660: Agent: ReadPipe: {"cmd":"serverstate","value":1,"url":"wss://remote.acmecomputer.com:443/agent.ashx","amt":false}
02:21:AM.0761: Agent_onStateChanged 1, 1, acmeComputerRemote
02:21:AM.0821: Agent: SendSelfUpdateQuery
02:21:AM.0821: Agent: ReadPipe: {"cmd":"sessions","sessions":{"terminal":{},"files":{},"tcp":{},"udp":{},"msg":{}}}
02:21:AM.8966: agentSelection_Click ~
02:21:AM.8966: connectToAgent ~
02:21:AM.8966: Agent: DisconnectPipe()
02:21:AM.9097: Agent_onStateChanged 0, 0, ~
```

**Comment by jobrien2001** [comment]
Adding some info.

Ran meshcentral with --debug... seems like i only get to the server when agent is selected to the monitor.  I get no responses  on direct connect.

**Comment by si458** [comment]
did you find a solution at all?
ive just noticed the same thing,
downloaded the assistant monitor only, but it then loads up as a standard client
i have to switch manually every time in the menu using agent select to Mesh Agent to get it to monitor the background service

**Comment by jobrien2001** [comment]
No solution, I gave up.

**Comment by si458** [comment]
this has been fixed recently,
i fixed it in 1.1.18, then realised a little bug, so have done another commit to fix it again https://github.com/Ylianst/MeshCentral/commit/0b0f2999db123fe48e43d2449beed676006407f1
if you use 1.1.18, downloading the assistant will ALWAYS launch it in monitor mode now until we release 1.1.19,
sorry about this
